tests/foreman/ui/test_contentview.py

```diff
@@ -1144,13 +1147,103 @@ def test_positive_promote_composite_with_custom_content(self):
 
         @steps: create a composite view containing multiple content types
 
-        @assert: Content view can be promoted
-
-        @caseautomation: notautomated
-
+        @assert: Composite content view can be promoted
 
         @CaseLevel: Integration
         """
+        env_name = gen_string('alpha')
+        cv1_name = gen_string('alpha')
+        cv2_name = gen_string('alpha')
+        cv_composite_name = gen_string('alpha')
+        custom_repo1_name = gen_string('alpha')
+        custom_repo2_name = gen_string('alpha')
+        custom_repo1_url = FAKE_0_YUM_REPO
+        custom_repo2_url = FAKE_1_YUM_REPO
+        puppet_repo1_url = FAKE_0_PUPPET_REPO
+        puppet_repo2_url = FAKE_1_PUPPET_REPO
+        puppet_module1 = 'httpd'
+        puppet_module2 = 'ntp'
```

Comment:
Requesting @svtkachenko look on these puppet_modules, as he has recently proposed numerous fixes and basically used different ones in the end.
Otherwise code looks good to me